samples/extensions/open_cl_interop/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021, Arm Limited and Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 the "License";
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+ 
+# The OpenCL interoperability only works on Android
+get_filename_component(FOLDER_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+get_filename_component(PARENT_DIR ${CMAKE_CURRENT_LIST_DIR} PATH)
+get_filename_component(CATEGORY_NAME ${PARENT_DIR} NAME)
+
+# OpenCL
+add_library(opencl INTERFACE)
+set(OPENCL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/OpenCL-Headers/)
+target_sources(opencl INTERFACE ${OPENCL_INCLUDE_DIR}/CL/opencl.h)
+target_include_directories(opencl INTERFACE ${OPENCL_INCLUDE_DIR})
+
+if(ANDROID)

```

Comment:
took me a bit to figure this out.  the problem with this path of limiting inclusion of the add_sample_with_tags for only one platform is if you're on a different platform and try to run the sample, or list samples, the folder discovery logic is still finding this sample.  In order to keep things working with other samples as normal, let's put the project specialization within the code of the sample itself.
This will promote both the idea of future work on making work with other platforms and allow for the system to correctly say "extension not found" rather than failing out in a unique way.